server/text_generation_server/models/vlm_causal_lm.py

```diff
@@ -39,15 +39,14 @@ def get_anyres_image_grid_shape(image_size, grid_pinpoints, patch_size):
     return height // patch_size, width // patch_size
 
 
-def image_text_replacement(image_input, config, image_id) -> str:
+def image_text_replacement(processor, image_input, config, image_id) -> str:
     if config.model_type == "idefics2":
         # TODO technically depends on image splitting which is not implemented.
-        num_features = 320
-        return (
-            "<fake_token_around_image>"
-            + "<image>" * num_features
-            + "<fake_token_around_image>"
-        )
+        image_seq_len = 64
+        image_str = f"<fake_token_around_image>{'<image>' * image_seq_len}<fake_token_around_image>"
+        if processor.image_processor.do_image_splitting:
+            image_str *= 5
```

Comment:
https://github.com/huggingface/transformers/blob/0dd65a03198424a41ec6948e445c313e9f292939/src/transformers/models/idefics2/processing_idefics2.py#L191-L193

We still need that.

Instead of doing it that inefficiently maybe we just conditionally prefix/postfix (Also to make the intent clearer). Wdyt ?